### PR TITLE
Gracefully handle parse errors

### PR DESCRIPTION
**Pull request by olafurpg**
This commit changes error handling for parsing source files so that
users get nicely formatted error messages instead of a large stack trace.

### CODE REVIEW

File: src/main/scala/com/github/tkawachi/doctest/ScaladocExtractor.scala [hunk 1]
```diff
@@ -12,8 +15,25 @@ object ScaladocExtractor {
...
+    val trimmedText = text.replace("\r\n", "\n")
+    val input = Input.VirtualFile(file.toString, trimmedText)
+    extractFromInput(input)
+  }
+  def extractFromInput(scalaSource: Input): List[ScaladocComment] = {
+    scalaSource.parse[Source] match {
+      case Parsed.Success(tree) =>
+        extractFromTree(tree)
+      case error @ Parsed.Error(_, _, _) =>
+        println(error.toString())
```
Comment (olafurpg) [inline src/main/scala/com/github/tkawachi/doctest/ScaladocExtractor.scala]: This results in `sbt test` passing when there are parse errors so we might want a better solution. I personally recommend having a `Reporter` class that can `.error(pos, message)` and also test for `.hasError`. I've been meaning to expose such functionality in Scalameta itself.

File: src/main/scala/com/github/tkawachi/doctest/ScaladocTestGenerator.scala [hunk 2]
```diff
@@ -17,9 +17,8 @@ object ScaladocTestGenerator {
    * Generates test source code from scala source file.
    */
   def apply(srcFile: File, srcEncoding: String, testGen: TestGen, decodeHtmlEnabled: Boolean, onlyCodeBlocksMode: Boolean): Seq[TestSource] = {
-    val src = Source.fromFile(srcFile, srcEncoding).mkString
```
Comment (olafurpg) [inline src/main/scala/com/github/tkawachi/doctest/ScaladocTestGenerator.scala]: I think `Source.fromFile(..).mkString` leaks file handles, it's better to use `java.nio.file.Files.readAllBytes` in my experience

File: src/main/scala/com/github/tkawachi/doctest/ScaladocExtractor.scala (hunk 1 above)
Comment (jozic) [reply src/main/scala/com/github/tkawachi/doctest/ScaladocExtractor.scala]: if parser can't handle it, doesn't it mean that compilation for that file fails? 
if so, then we don't even get to tests and as such to doctests

File: src/main/scala/com/github/tkawachi/doctest/ScaladocTestGenerator.scala (hunk 2 above)
Comment (jozic) [reply src/main/scala/com/github/tkawachi/doctest/ScaladocTestGenerator.scala]: :+1:

File: src/main/scala/com/github/tkawachi/doctest/ScaladocExtractor.scala (hunk 1 above)
Comment (olafurpg) [reply src/main/scala/com/github/tkawachi/doctest/ScaladocExtractor.scala]: Correct, the compilation would fail in that case.

**Comment by olafurpg** [comment]
I recommend Appveyor for integration tests on Windows. I propose to do that in a separate PR since it requires repo admin privileges to configure, here is an example appveyor.yml I use in scalafix

https://github.com/scalacenter/scalafix/blob/master/appveyor.yml

**Comment by jozic** [comment]
>I propose to do that in a separate PR

totally agree

**Comment by jozic** [comment]
@tkawachi do you want to review this?

@olafurpg this not only `Gracefully handles parse errors` but also fixes the issue with EOL on Windows, right?

**Comment by olafurpg** [comment]
>  this not only Gracefully handles parse errors but also fixes the issue with EOL on Windows, right?

@jozic correct

**Comment by tkawachi** [comment]
LGTM

**Comment by jozic** [comment]
@tkawachi do you mind releasing 0.9.3 with this fix?

**Comment by tkawachi** [comment]
@jozic yes, I'll release tomorrow, or this weekend.

**Pull request merged**